compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -481,7 +481,38 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                         }
                         if let Some(ref s) = note {
                             // If it has a custom `#[rustc_on_unimplemented]` note, let's display it
-                            err.note(s.as_str());
+                            if Some(trait_ref.def_id()) == tcx.lang_items().div_trait() {
```

Comment:
suggestion: Might as well look for `add_trait` or `sub_trait` or `mul_trait` too, while you're at it?

Reply:
I improved it in order to look for them!